Makefile

```diff
@@ -372,17 +372,17 @@ venv:
 ## Extraction step
 setup:
 	$(MAKE) -C tools
-	$(PYTHON) tools/buildtools/decompress_baserom.py $(VERSION)
+	$(PYTHON) tools/buildtools/decompress_baserom.py --mm-version $(VERSION)
 	$(PYTHON) tools/buildtools/extract_baserom.py $(BASEROM_DIR)/baserom-decompressed.z64 -o $(EXTRACTED_DIR)/baserom --dmadata-start `cat $(BASEROM_DIR)/dmadata_start.txt` --dmadata-names $(BASEROM_DIR)/dmadata_names.txt
-	$(PYTHON) tools/buildtools/extract_yars.py $(VERSION)
+	$(PYTHON) tools/buildtools/extract_yars.py --baserom-segments $(EXTRACTED_DIR)/baserom --mm-version $(VERSION)
```

Comment:
Same here and for the other scripts. I'm not a big fan of using explicit flags for required parameters. Feels counterintuitive

Reply:
tools in OoT take the version as --oot-version/-v and it's optional, defaulting to gc-eu-mq-dbg
For consistency with those I also used it even if mandatory
not sure how to reconcile

Reply:
I'll admit I don't think I have too strong opinion on using explicit flags for required arguments. I think I like the consistency just a little bit more so would probably prefer going that route. 
However I would say if the argument really is required, it should probably not have a default and actually require the user to specify it (`extract_baserom.py` in OoT for example doesn't have a default option).

Reply:
Made optional arguments positional